### DISCUSSION TITLE
Pass some env vars to executed commands

### DISCUSSION THREAD

**Discussion by betterphp** [streamdeck-linux-gui/streamdeck-linux-gui#104] [Ideas / Feature Requests]
Would it be possible to pass some environment variables to the commands executed by the buttons?

I want to make a script that interacts with the button that was pressed so the index of the button/page as an env vars would be handy.

**Comment by coolapso** [comment]
that's not really advisable. 
The best way is to create a scrip and handle all the env variables within the script. maybe accept some args on the script and handle the variables accordingly.  store it locally and then use the command feature to execute the script

**Reply by betterphp** [reply]
Thanks, how would I know which button was pressed in the script though?

**Reply by betterphp** [reply]
I'm trying to avoid hardcoding indexes in my scripts basically.

**Reply by rombert** [reply]
I was thinking about something similar these days and I would find it useful. Consider a script that toggles something, e.g. mute/unmute microphone. That script could then 'call back' and update the icon to reflect the execution results using `streamdeckc`.

In order to support any kind of layout it would need to know the button index, so the most straightforward way would be to set env vars like
- `STREAMDECK_UI_BUTTON_INDEX`
- `STREAMDECK_UI_DECK_INDEX`
- `STREAMDECK_UI_PAGE_INDEX`
when launching the script.

The problem with supplying these manually is that whenever I would drag the button around I would also need to update the scripts, which is error-prone.

**Reply by betterphp** [reply]
I ended up implementing what I needed using https://www.npmjs.com/package/@elgato-stream-deck/node

It was loads more effort but pretty fun to play with and seems to be able to update the button with nearly no delay which is nice.

**Comment by Quintar** [comment]
Bist have them as command like arguments which you add in the button
command line

Jacek ***@***.***> schrieb am Do., 19. Okt. 2023, 16:00:

**Reply by betterphp** [reply]
Yeah but is there a variable or something I could use in the command to be replaced with the button index? If I drag the button to a new location it'd be just as awkward to update the command as it would update my script.
